applications/TrilinosApplication/custom_python/add_trilinos_linear_solvers_to_python.cpp

```diff
@@ -129,6 +129,8 @@ void  AddLinearSolvers(pybind11::module& m)
         .def("Create",&TrilinosLinearSolverFactoryType::Create)
         .def("Has",&TrilinosLinearSolverFactoryType::Has)
         ;
+
+    RegisterTrilinosLinearSolvers();
```

Comment:
why do you move this?
Has nothing to do with python

Reply:
Linking problems during compilation

Reply:
Hm I am not sure if this is correct but I don't know
I am concerned that the linear solvers are not being registered if the python-part is not loaded, which could happen I guess. Especially I think this would be the use-case in the MappingApp once I use the linear solvers from trilinos

Lets wait for the feedback of @roigcarlo and @jcotela

Reply:
In any case as I said I still have problems with the linking when importing. I need someone to check  this

Reply:
Can you post the error?

Reply:
I posted in a different PR, wait...

Reply:
~~~sh
from KratosTrilinosApplication import *
ImportError: /home/vicente/src/Kratos/libs/KratosTrilinosApplication.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _ZN9ML_Epetra11SetDefaultsENSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEERN7Teuchos13ParameterListEPiPdb
~~~

No idea :(

Reply:
LD_PRELOAD does not work

Reply:
ah I think this is the same one you had some time ago?

are you compiling trilinos yourself?

Reply:
I tried that and the ubuntu ones, same error

Reply:
Honestly, I would not move this unless it solves a specific problem, since I share the concerns of @philbucher.